packages/calypso-e2e/src/lib/components/editor-settings-sidebar-component.ts

```diff
@@ -341,16 +337,9 @@ export class EditorSettingsSidebarComponent {
 	 * @param {string} slug URL slug to set.
 	 */
 	async enterUrlSlug( slug: string ) {
-		const inputLocator = this.editor.locator( selectors.permalinkInput );
-		await inputLocator.fill( slug );
-		// Hit the Tab key to confirm URL slug input and update the Post URL
-		// shown in this section.
-		await this.page.keyboard.press( 'Tab' );
-
-		const generatedURL = this.editor.locator(
-			`${ selectors.permalinkGeneratedURL } > text=/${ slug }`
-		);
-		await generatedURL.waitFor();
+		await this.editor.getByRole( 'button', { name: /Change URL:/ } ).click();
+		await this.editor.getByLabel( 'Permalink' ).fill( slug );
+		await this.editor.getByRole( 'button', { name: 'Close' } ).click();
```

Comment:
Nice use of the new role-based locators!

Reply:
[Needed to add `exact`](https://github.com/Automattic/wp-calypso/pull/70985/commits/33e5b488cfa5f56874673c8fbcf3ba999ade6d41) since it was added in v1.28. It wasn't failing for me locally because I was still on v1.27 🤷